Library/Homebrew/linkage_checker.rb

```diff
@@ -30,6 +31,7 @@ def check_dylibs
       # when checking for broken linkage
       file.dynamically_linked_libraries(except: :LC_LOAD_WEAK_DYLIB).each do |dylib|
         @reverse_links[dylib] << file
+        next if checked_dylibs.include? dylib
```

Comment:
This just eliminates checking the same `dylib` more than once, right?

Reply:
right. This check kicks in when different executables/libraries of the formula link to the `dylib` that has been already processed and added to the appropriate list.